tests/init/neg/cold-insert-hot-array.scala

```diff
@@ -0,0 +1,5 @@
+object A:
+  def foo[T](x: T, array: Array[T]): Unit = array(0) = x
+  var a = Array.apply(1, 2, 3)
+  foo(i, a)
+  val i = 99
```

Comment:
Here `i` is not cold. You can use e.g. `this` instead. It's better to only leave the method `foo` in the global object `A`, and move other code to a class --- this way, we can be sure that the method call `foo(...)` will be `A.foo(...)` instead of `A.this.foo(...)`.

Reply:
I was not clear enough in the above. I think something like the following will serve the purpose:

```Scala
class B {
  var a = new Array[B](2)
  A.foo(this, a)
  println(a(0).i)
  val i = 99
}
```